api/src/main/java/io/onetable/client/PerTableConfig.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package io.onetable.client;
+
+import java.util.List;
+
+import io.onetable.model.sync.SyncMode;
+
+public interface PerTableConfig {
```

Comment:
This interface appears to have little value. I don't think we need to change `PerTableConfig` for `ServiceLoader` integration. We can use the existing class to initialize the target clients for now.

Reply:
This was an open question in the filed issue. We either need to extract an interface, add a dependency from api on core or move PerTableConfig into api. If we are going to move to an injection model where the config doesn't need to be known by the clients - which I would suggest we do - then it can go completely away and even the use of the interface within init() could be removed. All of the interfaces that were extracted in this PR are in order to avoid moving implementation classes or adding dependencies to api module on core.

Reply:
IMO, moving `PerTableConfig` to api is cleaner than creating this interface. This also means that `HudiSourceConfig` and `IcebergCatalog` will also need to be moved to `api`.  To me this is better than creating the interfaces. I had created another issue to clean up the per table config, which could be used to track cleaning of these classes also. 

I am also supportive of dependency injection. Which framework did you have in mind?

Reply:
Interfaces are definitely cleaner for testing and api/spi modules but we can certainly move it. It shouldn't pull in other dependencies since I already extracted the other interfaces. I'll give it a shot.

Reply:
@ashvina - I don't have strong opinions about dependency injection frameworks here. I think rolling our own for this simple case would work just fine. If we moved to a more generic name-value pair config that would make it really easy.

I've tried to move the PerTableConfigImpl into api and like you said it will pull the Target specific configs along with it which is kinda messy. Do you want them in target specific sub-packages or just in the client package?

Personally, I think that the interface based approach that I did here to get ServiceLoader working is sufficient for this PR and the config refactoring could follow up and clean that up however appropriate. Mixing them in this PR complicates this work and will still need cleanup in the follow on anyway.

Thoughts on that?

Reply:
I don't like the idea of pulling the hudi and iceberg specifics into the api module. If we can avoid that, then that is preferred. We may need to do a larger refactor to do this though so I don't want to block progress for something that will probably require its own PR. I think we can try to pull each format's specifics into their own module in a future refactor and that can help us really find what is common and may help us structure this better.

Reply:
Yes, format specific in `api` is not a good design. The introduction of an interface for `PerTableConfig`, which also had format specific contract, did not address this design issue. Fixing this requires refactoring. I have started a discussion thread towards this #297

Reply:
Interpreting this as leave it as it is and move on within #297. Is that correct, @ashvina ?

Reply:
Hi. I apologize for the late reply. I had a slightly different understanding. The objective is to keep `api` free from table format specific configs. This interface does not address that and does not provide much benefit here. In fact, it causes many minor changes (for example, there are many `PerTableConfig` to `PerTableConfigImpl` changes in test cases).

I think we should remove the interface, just relocate the existing `PerTableConfig`.

Reply:
@ashvina  - This interface is necessary for the ServiceLoader to avoid dependency on implementation classes. We can't move the implementation class there as it will pull in other implementation classes due to how it defaults certain values. In which case, we would need to add additional implementation classes or a dependency on core. This interface only has class (interface and method) names that that are format specific but are not implementation classes which I imagined would be changed as part of the PerTableConfig refactoring.

The other option that I can see is that we put this in as is and we follow up with some change to move the implementation class into api that is decoupled from the format specific implementation classes. I'd actually like to see that refactoring result in quite the opposite and PerTableConfig not be in the api at all with injection removing the need for it in that layer.

Bottom line is that today PerTableConfig is dependent on format specific classes. The intent of this PR is not to fix that but to allow for ServiceLoader to work and one of the goals of #297 should indeed be to avoid those dependencies at least within the api layer.

Reply:
I appreciate your clarification. I have nothing else to do regarding this PR. However, I believe #297 should be our next priority.